### DISCUSSION TITLE
Improving Nuxt 3 Application Performance

### DISCUSSION THREAD

**Discussion by monkemedia** [nuxt/nuxt#16040] [Questions] [answered]
So I am doing everything possible to increase the performance of my website, and get as much GREEN as possible in Lighthouse.

To improve performance of a large application, is it better to turn off auto imports of components and import them directly into your page like so?

nuxt.config.js
```
export default defineNuxtConfig({
  components: {
    global: false,
  },
  ...
```
  
 index.vue
 ```
 <template>
 <div>
     ...
     <Instagram />
 </div>
 </template>
 
 <script setup>
     import Instagram from '@/components/Instagram'
 </script>
 ```
 
 If this is true how do we go about handling lazy components using the method above ^^^ ? From what i can see, the lazy component loads from the offset. 
 
 this is my current code
 
 ```
 <template>
     <button @click="() => test = true">Click</button>
     <LazyInstagram v-if="test"/>
 </template>
 
 <script setup>
     import LazyInstagram from '@/components/Instagram'
     
     const test = ref(false)
 </script>
 ```
 
 This is what i see when the page loads
 
![image](https://user-images.githubusercontent.com/14890432/200875512-da749fd8-877b-4930-8dea-666a3b4e3644.png)

Am I doing this correctly?

**Comment by TheAlexLichter** [comment, marked as answer]
AFAIK, Auto import should not have any impact on your app performance because everything that's not needed will be tree-shaken anyway.

**Reply by monkemedia** [reply]
That is very good to know. Many thanks.

Just need to find a plugin or something which will load the component when it comes into view. Think this will really improve performance

**Comment by bot08** [comment]
Nuxt does everything possible to make your vue application run as fast as possible. Therefore, it is better to pay attention to the lighthouse remarks and follow the remarks. You can also use SSG for faster server response.
